src/test/java/plugins/FolderPluginTest.java

```diff
@@ -167,6 +178,144 @@ public void folderScopedCredentialsTest() {
+
+        this.checkItemExists(job2, false);
+        this.checkItemExists(job2Renamed, true);
+        this.checkItemNameAndUrl(job2Renamed, JOB4_NAME, folder1, folder2);
+
+        job3.delete();
+
+        waitFor(driver, hasURL(folder2.url), 3);
+
+        this.checkItemExists(job3, false);
+
+        folder1.delete();
+
+        waitFor(driver, hasURL(jenkins.url), 3);
+        this.checkItemExists(folder1, false);
+        this.checkItemExists(job3, false);
+        this.checkItemExists(folder2, false);
+        this.checkItemExists(job2Renamed, false);
+    }
+
+    @Test
+    public void copyTest() {
+        final Folder folder1 = this.createFolder(FOLDER1_NAME);
+        final FreeStyleJob job1 = folder1.getJobs().create(FreeStyleJob.class, JOB1_NAME);
+        final Folder folder2 = this.createFolder(FOLDER2_NAME);
+        final Folder folder3 = this.createFolder(FOLDER3_NAME);
+
+        folder2.getJobs().copy(folder1.name + "/" + job1.name, job1.name);
+
+        final FreeStyleJob job2 = folder2.getJobs().get(FreeStyleJob.class, job1.name);
+        this.checkItemExists(job2, true);
+
+        folder3.getJobs().copy(folder2.name, folder2.name);
+
+        final Folder f3f2 = folder3.getJobs().get(Folder.class, folder2.name);
+        final FreeStyleJob job3 = f3f2.getJobs().get(FreeStyleJob.class, job1.name);
+        this.checkItemExists(job2, true);
+        this.checkItemExists(f3f2, true);
+        this.checkItemExists(job3, true);
+
+        try {
+            job3.startBuild();
+            fail("Copied job should not be able to build until saved");
+        } catch (final NoSuchElementException ex) {
+            Assert.assertThat(ex.getMessage(), containsRegexp("Build Now"));
+            Assert.assertThat(ex.getMessage(), containsRegexp("Unable to locate"));
+        }
+
+        job3.configure();
+        job3.save();
+        job3.startBuild().shouldSucceed();
+
+        f3f2.getJobs().copy("/" + folder1.name + "/" + job1.name, job1.name);
+
+        Assert.assertThat(driver, hasContent("A job already exists with the name ‘" +  job1.name + "’"));
+    }
+
+    private void checkItemNameAndUrl(final TopLevelItem item, final String itemName, final TopLevelItem... parentItems) {
+        item.open();
+        Assert.assertThat("Item name is not displayed", driver, hasContent(itemName));
+
+        if (parentItems.length > 0) {
+            String itemPath = "";
+
+            for (final TopLevelItem parentItem : parentItems) {
+                itemPath += String.format("%s/", parentItem.name);
+            }
+
+            itemPath += item.name;
+
+            Assert.assertThat("Complete path for item is not displayed", driver, hasContent(itemPath));
+        }
+    }
+
+    private void checkItemExists(final TopLevelItem f, final boolean expectedExists) {
+        boolean exists = true;
+
+        try {
+            IOUtils.toString(f.url("").openStream());
```

Comment:
```suggestion
            IOUtils.toString(f.url("").openStream(), StandardCharsets.UTF_8);
```
If applicable (import needed), if not, feel free to disregard 👏